contracts/AirDrop.sol

```diff
@@ -0,0 +1,113 @@
+pragma solidity 0.6.4;
+
+import "./interface/IBEP20.sol";
+import "./interface/ITokenHub.sol";
+import "./interface/IAirDrop.sol";
+import "./lib/SafeMath.sol";
+import "./System.sol";
+import "./MerkleProof.sol";
+
+contract AirDrop is IAirDrop, System {
+    using SafeMath for uint256;
+
+    string public constant sourceChainID = "Binance-Chain-Tigris"; // TODO: replace with the real chain id
+    address public constant approvalAddress = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa; // TODO: replace with the real address
+    bytes32 public constant override merkleRoot = 0xad4aa415f872123b71db5d447df6bb417fa72c6a41737a82fdb5665e3edaa7c3; // TODO: replace with the real merkle root
+
+    // This is a packed array of booleans.
+    mapping(uint256 => uint256) private claimedBitMap;
+
+    function isClaimed(bytes32 node) public view override returns (bool) {
+        uint256 index = uint256(node);
+        uint256 claimedWordIndex = index / 256;
+        uint256 claimedBitIndex = index % 256;
+        uint256 claimedWord = claimedBitMap[claimedWordIndex];
+        uint256 mask = (1 << claimedBitIndex);
+        return claimedWord & mask == mask;
+    }
+
+    function _setClaimed(bytes32 node) private {
+        uint256 index = uint256(node);
+        uint256 claimedWordIndex = index / 256;
+        uint256 claimedBitIndex = index % 256;
+        claimedBitMap[claimedWordIndex] = claimedBitMap[claimedWordIndex] | (1 << claimedBitIndex);
+    }
+
+
+    function claim(uint256 tokenIndex, bytes32 tokenSymbol, uint256 amount, bytes calldata ownerSignature, bytes calldata approvalSignature, bytes32[] calldata merkleProof) external override {
+        // Recover the owner address and check signature.
+        address ownerAddr = _verifyTMSignature(ownerSignature, keccak256(abi.encodePacked(sourceChainID, tokenSymbol, tokenIndex, amount, msg.sender)));
```

Comment:
for the signature schema of cosmos, it can not recover account from signature, you have to pass the account to verify by pubkey.

Reply:
updated